apps/api/src/app/events/e2e/trigger-event.e2e.ts

```diff
@@ -597,12 +597,18 @@ describe(`Trigger event - ${eventTriggerPath} (POST)`, function () {
       _environmentId: session.environment._id,
     });
 
-    await integrationRepository.delete({
-      _id: { $in: existingIntegrations.map((integration) => integration._id) },
+    const integrationIdsToDelete = existingIntegrations.flatMap((integration) =>
+      integration._environmentId === session.environment._id ? [integration._id] : []
+    );
+
+    const deletedIntegrations = await integrationRepository.deleteMany({
```

Comment:
Had to create a deleteMany method as the delete one has a safeguard of looking for one integration. With the change of paradigm and after adding test data for production environment too can't be relied on for this task.